middlewares/authenticator.go

```diff
@@ -88,6 +105,23 @@ func parserDigestUsers(users types.Users) (map[string]string, error) {
 	return userMap, nil
 }
 
+func getLinesFromFile(filename string) ([]string, error) {
+	dat, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	// Trim lines and filter out blanks
+	rawLines := strings.Split(string(dat), "\n")
+	filteredLines := rawLines[:0]
```

Comment:
Has something to do with the array backing the first slice being reused for the second slice... I'll be honest, it sounds suspicious.